sklearn/utils/estimator_checks.py

```diff
@@ -2501,3 +2514,42 @@ def check_fit_idempotent(name, estimator_orig):
         if hasattr(estimator, method):
             new_result = getattr(estimator, method)(X_test)
             assert_allclose_dense_sparse(result[method], new_result)
+
+
+def check_outlier_rejectors(name, estimator_orig):
+    X, y = make_blobs(random_state=0)
+    outliers = estimator_orig.fit_predict(X, y) == -1
+    n_outliers = np.sum(outliers)
+
+    X_new, y_new = estimator_orig.fit_resample(X, y)
+
+    assert X_new.shape[0] == X.shape[0] - n_outliers
+    assert y_new.shape[0] == y.shape[0] - n_outliers
+
+
+def check_resampler_structure(name, estimator_orig):
+    X, y = make_blobs(n_samples=10)
+    X_new, y_new = estimator_orig.fit_resample(X, y)
```

Comment:
we should allow kwargs to be returned also (optionally, or required, I'm not sure)

Reply:
Assuming X is array-like, X_new should have the same number of features and the same dtype.

Reply:
Are there any notable kwargs we need to handle other than `sample_weight`?

Reply:
> Are there any notable kwargs we need to handle other than sample_weight?

Not sure, but I don't see any reason to restrict it to `sample_weight`.